src/edu/csus/ecs/pc2/ui/ShadowControlPane.java

```diff
@@ -503,6 +546,54 @@ private ShadowInformation getCurrentShadowInformation(ContestInformation contest
         return newShadowInfo;
     }
 
+    private void setupConnectionStatusTable() {
+
+        Object[] columns = { "Time             ", "Description               " };
+        connectStatusTable.removeAll();
+        
+        connectStatusTableModel = new DefaultTableModel(columns, 0) {
+            @Override
+            public boolean isCellEditable(int row, int col) {
+                return false;
+            }
+        };
+
+        connectStatusTable.setModel(connectStatusTableModel);
+
+        // Sorters
+        TableRowSorter<DefaultTableModel> trs = new TableRowSorter<DefaultTableModel>(connectStatusTableModel);
+        
+        connectStatusTable.setRowSorter(trs);
+        connectStatusTable.setAutoResizeMode(JTable.AUTO_RESIZE_OFF);
+        
+        ArrayList<SortKey> sortList = new ArrayList<SortKey>();
+        
+        /*
+         * Column headers left justified
+         */
+        ((DefaultTableCellRenderer)connectStatusTable.getTableHeader().getDefaultRenderer()).setHorizontalAlignment(JLabel.LEFT);
+        connectStatusTable.setRowHeight(connectStatusTable.getRowHeight() + VERT_PAD);
+                     
+        int idx = 0;
+
+        // These are in sort order
+        // Time
+        sortList.add(new RowSorter.SortKey(idx++, SortOrder.ASCENDING));
```

Comment:
The ability to sort the table rows by time is very useful.  However, after testing it I found it confusing because when you click on the column header to reverse the sort order, the vertical scrollbar does not also adjust.  For example, if you have multiple rows (enough to cause a Scrollbar to appear) and you click the column so that the sort order is "lowest time first", the time which appears at the top of the view is _not necessarily the lowest time_  -- it's only the lowest time _within the view indicated by the current position of the scrollbar_.  It seems to me that it would be better if, when the user clicks to "sort lowest to highest" the scrollbar would also adjust (to the top) so that it's clear that the top row _in the view_ is indeed the "lowest"  (and similarly for sorting "highest to lowest" -- I would expect the view in that case to show the highest value at the top).

I grant that other people may have a different preference (may not want the scrollbar to change when a sort is invoked), but it sure seems confusing to me as it is currently.

Reply:
Pretty much every table we have operates this way (take Accounts and Runs as an example).  I'm not saying it's correct, but, as you indicate it's a preference thing.  The "Windows" solution is to always jump to the "top" after any sort (in explorer) no matter where you were (if no item is selected.)   If an item is selected, it "makes sure that item is visible" after the sort.   We can do that (if an item is selected, make sure it's visible after the sort), but (and I cringe a little to say this),that's out of scope for this issue and if we do that, ALL tables should be changed to work the same way, which, we can probably do somewhat easily since all JTable's use JTableCustomized as a base class...

Reply:
I agree that we can consider this "out of scope".